src/Hook.php

```diff
@@ -0,0 +1,240 @@
+
+    /**
+     * Add Description
+     *
+     * @param string $desc
+     * @return static
+     */
+    public function desc(string $desc): static
+    {
+        $this->desc = $desc;
+
+        return $this;
+    }
+
+    /**
+     * Get Description
+     *
+     * @return string
+     */
+    public function getDesc(): string
+    {
+        return $this->desc;
+    }
+
+    /**
+     * Add Group
+     *
+     * @param array $groups
+     * @return static
+     */
+    public function groups(array $groups): static
+    {
+        $this->groups = $groups;
+
+        return $this;
+    }
+
+    /**
+     * Get Groups
+     *
+     * @return array
+     */
+    public function getGroups(): array
+    {
+        return $this->groups;
+    }
+
+    /**
+     * Add Action
+     *
+     * @param callable $action
+     * @return static
+     */
+    public function action(callable $action): static
+    {
+        $this->action = $action;
+        return $this;
+    }
+
+    /**
+     * Get Action
+     *
+     * @return callable
+     */
+    public function getAction()
+    {
+        return $this->action;
+    }
+
+    /**
+     * Get Injections
+     *
+     * @return array
+     */
+    public function getInjections(): array
+    {
+        return array_keys($this->injections);
+    }
+
+    /**
+     * Inject
+     *
+     * @param string $injection
+     *
+     * @throws Exception
+     *
+     * @return static
+     */
+    public function inject(string $injection): static
+    {
+        if (array_key_exists($injection, $this->injections)) {
+            throw new Exception('Injection already declared for ' . $injection);
+        }
+
+        $this->injections[$injection] = [
+            'name' => $injection,
+            'order' => count($this->params) + count($this->injections),
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Add Param
+     *
+     * @param string $key
+     * @param mixed $default
+     * @param Validator|callable $validator
+     * @param string $description
+     * @param bool $optional
+     * @param array $injections
+     *
+     * @return static
+     */
+    public function param(string $key, mixed $default, Validator|callable $validator, string $description = '', bool $optional = false, array $injections = []): static
+    {
+        $this->params[$key] = [
+            'default' => $default,
+            'validator' => $validator,
+            'description' => $description,
+            'optional' => $optional,
+            'injections' => $injections,
+            'value' => null,
+            'order' => count($this->params) + count($this->injections),
+        ];
+
+        return $this;
+    }
+
+    /**
+     * Get Params
+     *
+     * @return array
+     */
+    public function getParams(): array
+    {
+        return $this->params;
+    }
+
+    /**
+     * Get Param Values
+     *
+     * @return array
+     */
+    public function getParamsValues(): array
```

Comment:
```suggestion
    public function getParamValues(): array
```

Reply:
I just copied this form `Route` its actually we are getting the values of `params` property. so function name `getParamsValues` does make sense?

Reply:
I think it should be fine.